pkg/provider/aws/auth/auth.go

```diff
@@ -122,6 +122,9 @@ func New(ctx context.Context, store esv1beta1.GenericStore, kube client.Client,
 	}
 
 	for _, aRole := range prov.AdditionalRoles {
+		// TODO: delete this comment later
+		// assumeRoler calls AssumeRoleFunc, which is where
+		// the assertion for the test lives and currently fails
```

Comment:
The test failing is actually a good sign if we consider the Test Driven Development approach  - you have created a test that fails because you're not yet using the `ExternalID` field 😄